File: src/bin/0383-ransom-note.py
# @leetup=info id=383 lang=python3 slug=ransom-note



# @leetup=code

from collections import Counter


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        a = Counter(ransomNote)
        b = Counter(magazine)
        return all(a[k] <= b[k] for k in a)


# @leetup=code

File: src/bin/0383-ransom-note.cpp
// @leetup=info id=383 lang=cpp slug=ransom-note

#include <algorithm>
#include <iostream>
#include <map>
#include <set>
#include <tuple>
#include <vector>

// @leetup=code

class Solution {
 public:
  bool canConstruct(std::string ransomNote, std::string magazine) {
    std::map<char, int> a;
    for (char c : ransomNote) {
      a[c] += 1;
    }
    std::map<char, int> b;
    for (char c : magazine) {
      b[c] += 1;
    }
    for (auto it : a) {
      if (b[it.first] < it.second) {
        return false;
      }
    }
    return true;
  }
};
// @leetup=code

int main() {
  std::cout << "Hello, world!" << std::endl;
  return 0;
}
